R/utils.R

```diff
@@ -86,3 +86,83 @@ check_date_time <- function(date_time, date_name = "start_date") {
   )
   as.character(parsed)
 }
+
+#' Get the credentials from environment variables, or set them manually
+#'
+#' By default, it's not necessary to set any values in this function as it's
+#' used in the background by other functions. However, if you wish to provide
+#' your username and password on a per function basis, this function allows you
+#' to do so.
+#'
+#' @param username ETN Data username, by default read from the environment, but
+#'   you can set it manually too.
+#' @param password ETN Data password, by default read from the environment, but
+#'   you can set it manually too.
+#'
+#' @return A string as it is ingested by other functions that need
+#'   authentication
+#' @family helper functions
+#' @noRd
+
+get_credentials <-
+  function(username = Sys.getenv("userid"),
+           password = Sys.getenv("pwd")) {
+    stringr::str_glue('list(username = "{username}", password = "{password}")')
+  }
+
+#' Extract the OCPU temp key from a response object
+#'
+#' When posting a request to the opencpu api service without the json flag, a
+#' response object is returned containing all the generated objects, with a
+#' unique temp key in the path. To retrieve these objects in a subsequent GET
+#' request, it is convenient to retrieve this temp key from the original
+#' response object
+#'
+#' @param response The response resulting from a POST request to a opencpu api
+#'   service
+#'
+#' @return the OCPU temp key to be used as part of a GET request to an opencpu
+#'   api service
+#' @family helper functions
+#' @noRd
+extract_temp_key <- function(response) {
+  response %>%
+    httr::content(as = "text") %>%
+    stringr::str_extract("(?<=tmp\\/).{15}(?=\\/)")
+}
+
+#' Retrieve the result of a function called to the opencpu api
+#'
+#' Loading the evaluated object into the current environment, to be used
```

Comment:
I don't understand this long sentence. 😄

Reply:
### Fetch the result of an API call to OpenCPU

This function is used internally to GET an evaluated object from an OpenCPU api, to GET a result, you must of course POST it first